share/lib/python/neuron/__init__.py

```diff
@@ -447,6 +459,8 @@ def run(tstop):
     for your model.
 
     """
+    warnings.warn("neuron.run(tstop) is deprecated; use h.init() and h.continuerun(tstop) instead", DeprecationWarning)
```

Comment:
h.init() and h.continuerun(tstop) don't exist without either ```from neuron import gui``` or ```h.load_file("stdrun.hoc")```.
And to muddy the waters further, h may exist only as neuron.h in the context of being able to use ```neuron.run(tstop)```. Also, for h.init, do you mean h.stdinit() with its implicit global arg (v_init) or h.finitialize(...) itself. At the same time one can't put the whole world into a warning message so I'm conflicted about how much context to assume on the user's part. I'm in the habit of using ```from neuron import h, gui``` If I need a run() integrated with the existing gui. But that brings in tstop and dt issues. I'm kind of sorry I started this comment since, as is, neuron.run() to my mind is next to useless.

Reply:
Ugh. Fair point. I usually advise people to not import the GUI unless they're planning on using the GUI, and to just load stdrun.hoc if that's what they need... in the course I suggest loading `stdrun`, then an `finitialize`, then `continuerun`... I don't like `stdint` or `run` because of their dependence on global variables